impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultDependencyResolverResult.java

```diff
@@ -107,14 +127,14 @@ public class DefaultDependencyResolverResult implements DependencyResolverResult
      * @param root the root node of the dependency graph
      * @param count estimated number of dependencies
      */
-    public DefaultDependencyResolverResult(
+    DefaultDependencyResolverResult(
             DependencyResolverRequest request,
             PathModularizationCache cache,
             List<Exception> exceptions,
             Node root,
             int count) {
         this.request = request;
-        this.cache = cache;
+        this.cache = Objects.requireNonNull(cache);
```

Comment:
or here 
```suggestion
        this.cache = Objects.requireNonNull(moduleCache());
```
one of the suggestions is right. as `moduleCache()` is SSOT.

Reply:
The `moduleCache()` method is defined in another class. We cannot access that method from here.

However, this is indeed the kind of code that we would have after we clarified how to manage session-wide caches. The `moduleCache()` method would be in the class managing those caches. We are just not there yet.

Reply:
Fwiw, session wide informations can be stored in `Session.getData()`.